plugins/offboard/offboard_impl.cpp

```diff
@@ -72,6 +83,16 @@ Offboard::Result OffboardImpl::stop() const
 
 void OffboardImpl::start_async(Offboard::result_callback_t callback)
 {
+    {
+        //std::lock_guard<std::mutex> lock(_mutex);
+
+        if (_mode == Mode::NOT_ACTIVE) {
+            if (callback) {
+                callback(Offboard::Result::NO_SETPOINT_SET);
+            }
+        }
+    }
+
```

Comment:
Shouldn't this return here so that the following code isn't executed?

Reply:
Very good catch, thank you!